awx/api/serializers.py

```diff
@@ -3094,32 +3134,28 @@ def build_field(self, field_name, info, model_class, nested_depth):
                                              self.credential)
         return super(WorkflowJobTemplateNodeSerializer, self).build_field(field_name, info, model_class, nested_depth)
 
+    def build_relational_field(self, field_name, relation_info):
+        field_class, field_kwargs = super(WorkflowJobTemplateNodeSerializer, self).build_relational_field(field_name, relation_info)
+        # workflow_job_template is read-only unless creating a new node.
+        if self.instance and field_name == 'workflow_job_template':
+            field_kwargs['read_only'] = True
+            field_kwargs.pop('queryset', None)
+        return field_class, field_kwargs
+
     def validate(self, attrs):
         deprecated_fields = {}
         if 'credential' in attrs:  # TODO: remove when v2 API is deprecated
             deprecated_fields['credential'] = attrs.pop('credential')
         view = self.context.get('view')
-        if self.instance is None and ('workflow_job_template' not in attrs or
-                                      attrs['workflow_job_template'] is None):
-            raise serializers.ValidationError({
-                "workflow_job_template": _("Workflow job template is missing during creation.")
-            })
+        attrs = super(WorkflowJobTemplateNodeSerializer, self).validate(attrs)
+        ujt_obj = None
         if 'unified_job_template' in attrs:
             ujt_obj = attrs['unified_job_template']
-        ujt_obj = None
-        if self.instance:
+        elif self.instance:
             ujt_obj = self.instance.unified_job_template
         if isinstance(ujt_obj, (WorkflowJobTemplate, SystemJobTemplate)):
             raise serializers.ValidationError({
                 "unified_job_template": _("Cannot nest a %s inside a WorkflowJobTemplate") % ujt_obj.__class__.__name__})
-        attrs = super(WorkflowJobTemplateNodeSerializer, self).validate(attrs)
-        if ujt_obj is None:
-            ujt_obj = attrs.get('unified_job_template')
-        accepted, rejected, errors = ujt_obj._accept_or_ignore_job_kwargs(**self._build_mock_obj(attrs).prompts_dict())
-        # Do not raise survey validation errors
-        errors.pop('variables_needed_to_start', None)
```

Comment:
This was a very bad method of excluding "variable x is missing" type errors. It also swept up other errors that we did not want to suppress like "variable x is too long". This is now accomplished by `exclude_errors`.